src/string.cpp

```diff
@@ -510,6 +513,16 @@ size_t Utf8Encode(char *buf, WChar c)
 	return 1;
 }
 
+size_t Utf8Encode(char *buf, WChar c)
+{
+	return Utf8Encode<char *>(buf, c);
+}
+
+size_t Utf8Encode(std::ostreambuf_iterator<char> &buf, WChar c)
+{
+	return Utf8Encode<std::ostreambuf_iterator<char> &>(buf, c);
+}
```

Comment:
Why are these specializations necessary? I'd usually think the regular template/overloading resolution would pick be able to resolve the templated Utf8Encode function. If there is a reason, I'd like a comment about it.

Reply:
The call resolves to recursively call itself otherwise, non-template functions bind  always bind over template functions.

Reply:
No I mean, why do those two non-template overloads exist? Shouldn't the template version be enough?

Reply:
I didn't want it and the whole ostream stuff included in the string header, as that is used in many files. It's a compile time optimisation.